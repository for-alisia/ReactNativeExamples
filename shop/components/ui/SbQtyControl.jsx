import React from 'react';
import { View, StyleSheet } from 'react-native';
import { AntDesign } from '@expo/vector-icons';

// Components
import SbText from './SbText';
import SbIconContainer from './SbIconContainer';

// Theme
import theme from '../../theme';

const SbQtyControl = ({ qty, onAdd, onSubstract }) => {
  return (
    <View style={styles.container} testID="control-ui">
      <SbIconContainer width={24} height={24} onPress={onSubstract} testID="control-minus">
        <AntDesign name="minus" size={16} color={theme.colors.greyLight} />
      </SbIconContainer>
      <SbText style={styles.text}>{qty}</SbText>
      <SbIconContainer onPress={onAdd} width={24} height={24} testID="control-plus">
        <AntDesign name="plus" size={16} color={theme.colors.greyLight} />
      </SbIconContainer>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    minWidth: 100,
    paddingHorizontal: theme.padding.m,
    paddingVertical: theme.padding.xs,
    borderColor: theme.colors.greyLight,
    borderWidth: 1,
    borderRadius: theme.borderRadius.m,
  },
  text: {
    color: theme.colors.greyLight,
  },
});

export default SbQtyControl;
