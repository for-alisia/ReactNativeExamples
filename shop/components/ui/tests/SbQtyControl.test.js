// @ts-nocheck
// Dependencies
import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';

// Component to test
import SbQtyControl from '../SbQtyControl';

describe('SbQtyControl UI Component', () => {
  const testQty = 5;
  const mockOnAdd = jest.fn();
  const mockOnSubstract = jest.fn();

  it('renders correctly with passed quantity', () => {
    const { getByText, getByTestId, toJSON } = render(<SbQtyControl qty={testQty} />);
    const text = getByText(testQty.toString());
    const element = getByTestId('control-ui');
    const minus = getByTestId('control-minus');
    const plus = getByTestId('control-plus');

    expect(text).toBeTruthy();
    expect(element).toBeTruthy();
    expect(minus).toBeTruthy();
    expect(plus).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls appropriate handlers', () => {
    const { getByTestId } = render(
      <SbQtyControl qty={testQty} onAdd={mockOnAdd} onSubstract={mockOnSubstract} />
    );
    const minus = getByTestId('control-minus');
    const plus = getByTestId('control-plus');

    fireEvent.press(minus);
    fireEvent.press(plus);

    expect(mockOnAdd).toHaveBeenCalledTimes(1);
    expect(mockOnSubstract).toHaveBeenCalledTimes(1);

    fireEvent.press(minus);
    fireEvent.press(plus);

    expect(mockOnAdd).toHaveBeenCalledTimes(2);
    expect(mockOnSubstract).toHaveBeenCalledTimes(2);
  });
});
