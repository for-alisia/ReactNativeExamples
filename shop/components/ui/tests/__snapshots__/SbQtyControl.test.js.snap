// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SbQtyControl UI Component renders correctly with passed quantity 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderColor": "#ADA19B",
      "borderRadius": 16,
      "borderWidth": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "minWidth": 100,
      "paddingHorizontal": 12,
      "paddingVertical": 4,
    }
  }
  testID="control-ui"
>
  <View
    style={
      Object {
        "borderRadius": 12,
        "height": 24,
        "width": 24,
      }
    }
    testID="control-minus"
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 100,
          "opacity": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
  <Text
    style={
      Object {
        "color": "#ADA19B",
        "fontFamily": "montserrat-regular",
        "fontSize": 16,
      }
    }
  >
    5
  </Text>
  <View
    style={
      Object {
        "borderRadius": 12,
        "height": 24,
        "width": 24,
      }
    }
    testID="control-plus"
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 100,
          "opacity": 1,
          "overflow": "hidden",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 24,
            "justifyContent": "center",
            "width": 24,
          }
        }
      >
        <Text />
      </View>
    </View>
  </View>
</View>
`;
